volmdlr/edges.py

```diff
@@ -543,54 +792,34 @@ def sort_points_along_line(self, points: List[volmdlr.Point2D]) -> List[
         return list_points
 
 
-class BSplineCurve2D(Edge):
+class BSplineCurve2D(BSplineCurve):
     _non_serializable_attributes = ['curve']
 
     def __init__(self,
                  degree: int,
                  control_points: List[volmdlr.Point2D],
                  knot_multiplicities: List[int],
                  knots: List[float],
-                 weights=None, periodic=False, name=''):
+                 weights: List[float] = None,
+                 periodic: bool = False,
+                 name: str = ''):
+
         self.control_points = control_points
         self.degree = degree
         knots = standardize_knot_vector(knots)
         self.knots = knots
         self.knot_multiplicities = knot_multiplicities
         self.weights = weights
         self.periodic = periodic
+        self.name = name
 
-        curve = BSpline.Curve()
-        curve.degree = degree
-        if weights is None:
-            P = [(control_points[i][0], control_points[i][1]) for i in
-                 range(len(control_points))]
-            curve.ctrlpts = P
-        else:
-            Pw = [(control_points[i][0] * weights[i],
-                   control_points[i][1] * weights[i], weights[i]) for i in
-                  range(len(control_points))]
-            curve.ctrlptsw = Pw
-        knot_vector = []
-        for i, knot in enumerate(knots):
-            knot_vector.extend([knot] * knot_multiplicities[i])
-        curve.knotvector = knot_vector
-
-        self.curve = curve
-        start = self.point_at_abscissa(0.)
-        end = self.point_at_abscissa(self.length())
-
-        Edge.__init__(self, start, end, name=name)
-
-    @classmethod
-    def from_geomdl_curve(cls, curve):
-        knots = list(sorted(set(curve.knotvector)))
-        knot_multiplicities = [curve.knotvector.count(k) for k in knots]
-        return BSplineCurve2D(degree=curve.degree,
-                              control_points=[volmdlr.Point2D(p[0], p[1]) for p in curve.ctrlpts],
-                              knots=knots,
-                              knot_multiplicities=knot_multiplicities
-                              )
+        BSplineCurve.__init__(self, self.degree,
+                              self.control_points,
+                              self.knot_multiplicities,
+                              self.knots,
+                              self.weights,
+                              self.periodic,
+                              self.name)
```

Comment:
You don't have to state `self.attribute = attribute` if the exact same statement is made in the inheritance class. 
For exemple : when you have a class inheriting from DessiaObject, you don't state `self.name = name` since the DessiaObject class does exactly that.

Same think for BSplineCurve3D